stress/src/main.rs

```diff
@@ -378,260 +588,51 @@ impl EventListener for WrittenBytesHook {
-    }
-    if let Some(s) = matches.value_of("batch_compression_threshold") {
-        config.batch_compression_threshold = ReadableSize::from_str(s).unwrap();
-    }
-    // Test configurations
-    if let Some(s) = matches.value_of("time") {
-        args.time = Duration::from_secs(s.parse::<u64>().unwrap());
-    }
-    if let Some(s) = matches.value_of("regions") {
-        args.regions = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("purge_interval") {
-        args.purge_interval = Duration::from_secs(s.parse::<u64>().unwrap());
-    }
-    if let Some(s) = matches.value_of("compact_ttl") {
-        args.compact_ttl = Duration::from_secs(s.parse::<u64>().unwrap());
-        if args.compact_ttl.as_millis() > 0 {
-            println!("Not supported");
-            std::process::exit(1);
-        }
-    }
-    if let Some(s) = matches.value_of("compact_count") {
-        args.compact_count = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("force_compact_factor") {
-        args.force_compact_factor = s.parse::<f32>().unwrap();
-    }
-    if let Some(s) = matches.value_of("write_threads") {
-        args.write_threads = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("write_ops_per_thread") {
-        args.write_ops_per_thread = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("read_threads") {
-        args.read_threads = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("read_ops_per_thread") {
-        args.read_ops_per_thread = s.parse::<u64>().unwrap();
-    }
-    if let Some(s) = matches.value_of("entry_size") {
-        args.entry_size = s.parse::<usize>().unwrap();
-    }
-    if let Some(s) = matches.value_of("write_entry_count") {
-        args.write_entry_count = s.parse::<u64>().unwrap();
+    config.dir = opts.path;
+    config.target_file_size = ReadableSize::from_str(&opts.target_file_size).unwrap();
+    config.purge_threshold = ReadableSize::from_str(&opts.purge_threshold).unwrap();
+    config.purge_rewrite_threshold =
+        Some(ReadableSize::from_str(&opts.purge_rewrite_threshold).unwrap());
+    config.purge_rewrite_garbage_ratio = opts.purge_rewrite_garbage_ratio.parse::<f64>().unwrap();
+    config.batch_compression_threshold =
+        ReadableSize::from_str(&opts.batch_compression_threshold).unwrap();
+    args.time = Duration::from_secs(opts.time.parse::<u64>().unwrap());
+    args.regions = opts.regions.parse::<u64>().unwrap();
+    args.purge_interval = Duration::from_millis(opts.purge_interval.parse::<u64>().unwrap());
+
+    if let Some(ttl) = opts.compact_ttl {
+        args.compact_ttl = Duration::from_millis(ttl.parse::<u64>().unwrap());
     }
-    if let Some(s) = matches.value_of("write_region_count") {
-        args.write_region_count = s.parse::<u64>().unwrap();
+
+    if args.compact_ttl.as_millis() > 0 {
+        println!("Not supported");
```

Comment:
This can be put into validator as well.

Reply:
I am a little confused about this validation. it seems that `compact_ttl` should always be zero . Should we just remove it from stress's paramters?

Reply:
Yes, it's a little hard to implement. You can remove it for now.